app/navigation/CategorisationNavigator.scala

```diff
@@ -346,16 +366,25 @@ class CategorisationNavigator @Inject() (categorisationService: CategorisationSe
 
   private def shouldGoToLongerCommodityCodeWhenCategory2(
     categorisationInfo: CategorisationInfo,
-    assessmentQuestion: CategoryAssessment
-  ) =
-    categorisationInfo.getMinimalCommodityCode.length == 6 && categorisationInfo.descendantCount != 0 && assessmentQuestion.category == Category2AsInt
+    assessmentQuestion: CategoryAssessment,
+    hasLongComCode: Boolean
+  ): Boolean =
+    !hasLongComCode &&
+      categorisationInfo.getMinimalCommodityCode.length == 6 &&
+      categorisationInfo.descendantCount != 0 &&
+      assessmentQuestion.category == Category2AsInt
 
   private def shouldGoToLongerCommodityCodeWhenCategory1(
     categorisationInfo: CategorisationInfo,
-    assessmentQuestion: CategoryAssessment
-  ) =
-    categorisationInfo.getMinimalCommodityCode.length == minimumLengthOfCommodityCode && categorisationInfo.descendantCount != 0 && assessmentQuestion.category == Category1AsInt && !categorisationInfo.categoryAssessmentsThatNeedAnswers
-      .exists(_.isCategory2) && categorisationInfo.categoryAssessments.exists(_.isCategory2)
+    assessmentQuestion: CategoryAssessment,
+    hasLongComCode: Boolean
+  ): Boolean =
+    !hasLongComCode &&
+      categorisationInfo.getMinimalCommodityCode.length == minimumLengthOfCommodityCode &&
+      categorisationInfo.descendantCount != 0 &&
+      assessmentQuestion.category == Category1AsInt &&
```

Comment:
Can we move the first three lines here into its own method? It looks duplicated in the method above so it could be worth removing that duplication if it will consistently be the same for both cat 1 and cat 2